src/structure/shapes.ts

```diff
@@ -23,30 +24,71 @@ export interface XYZ {
     z: number;
 }
 
-export interface SphereData {
-    kind: 'sphere';
+export interface BaseShapeData {
+    position: [number, number, number];
+    orientation?: [number, number, number, number];
+    scale?: number;
+    color?: ColorSpec;
+}
+
+export interface BaseShapeParameters<T> {
+    kind: string;
+    parameters: {
+        global: Partial<T>;
+        structure?: Array<Partial<T>>;
+        atom?: Array<Partial<T>>;
+    };
+}
+
+export interface SphereData extends BaseShapeData {
     radius: number;
 }
 
-// Interface for ellipsoidal data, where
-// orientation is stored in the (w, x, y, z) convention
-export interface EllipsoidData {
-    kind: 'ellipsoid';
+export interface SphereParameters extends BaseShapeParameters<SphereData> {
+    kind: 'sphere';
+}
+
+// Interface for ellipsoidal data, where the shape
+// of the ellipsoid is given by the semiaxes, and the orientation
+// by the general `orientation` property
+export interface EllipsoidData extends BaseShapeData {
     semiaxes: [number, number, number];
-    orientation?: [number, number, number, number];
+}
+
+export interface EllipsoidParameters extends BaseShapeParameters<EllipsoidData> {
+    kind: 'ellipsoid';
+}
+
+// Interface for arrow data
+export interface ArrowData extends BaseShapeData {
+    vector: [number, number, number];
+    base_radius?: number;
+    head_radius?: number;
+    head_length?: number;
```

Comment:
```suggestion
    baseRadius?: number;
    headRadius?: number;
    headLength?: number;
```

We should use JS style here